hs_core/migrations/0024_custom_migration_metadata_namespace_20160527.py

```diff
@@ -0,0 +1,34 @@
+from __future__ import unicode_literals
+
+import logging
+
+from django.db import migrations
+
+from hs_core.models import BaseResource
+from hs_core.hydroshare.utils import resource_modified
+
+def migrate_namespace_for_source_and_relation(apps, schema_editor):
+    # migrate the namespace for the 'Source' and 'Relation' metadata
+    # elements from 'dcterms' to 'hsterms' by regenerating the metadata xml
+    log = logging.getLogger()
+    for res in BaseResource.objects.all():
+        # need to regenerate xml file only for those resources that have either 'source' or 'relation' metadata
+        if len(res.metadata.sources.all()) > 0 or len(res.metadata.relations.all()) > 0:
+            resource_modified(res, res.creator)
+            log_msg = "Namespace for either 'source' or 'relation' metadata element was updated for resource " \
```

Comment:
@pkdash Although this custom migration looks to do its work, the migration could run unnecessarily if migration table newly generated did not get written to the database and hydroshare is redeployed with the original database. Since this migration just regenerates resourcemetadata.xml by calling resource_modified(), I guess it is OK. But if it is not difficult to check resourcemetadata.xml file stored in iRODS to make sure custom migration needs to run, i.e., namespace has not been updated, that would be ideal to prevent unnecessary migrations. I know it is extra work, so I am also debating whether it is worth the extra work to prevent unnecessary migrations. Bringing @gantian127 in the loop since she has been working on raster custom migration for similar issues. If you and @gantian127 decides it is not worth the effort to check data on irods side to prevent unnecessary custom migration runs, Let me know and I can give a +1. We just need to keep this in mind during deployment. For custom migrations that could result in corrupt resource content files potentially caused by multiple custom migration runs, we do need to make custom migrations more robust so as to make sure it only runs once.